capstone/src/main/java/com/google/univiz/api/representation/Deadline.java

```diff
@@ -1,10 +1,45 @@
 package com.google.univiz.api.representation;
 
 import com.google.auto.value.AutoValue;
+import com.ryanharter.auto.value.gson.GenerateTypeAdapter;
 
 /**
  * Contains the date when applications are due for a particular college. Will be made serializable
  * for the purpose of graphing due dates.
  */
 @AutoValue
-public abstract class Deadline {}
+@GenerateTypeAdapter
+public abstract class Deadline {
+  public abstract int openingMonth();
+
+  public abstract int openingDay();
+
+  public abstract int openingYear();
```

Comment:
In general, it's good to leverage existing Date / Time libraries where possible.

Can you try modeling this as a pair of [LocalDate](https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/time/LocalDate.html) instances? We'll also need to verify that GSON can serialize them...

Reply:
I was planning on using an existing datatype for this, but was worried about GSON not being able to serialize it. Further I wasn't sure if it would deserialize into the [Date](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date) datatype or if it would just end up deserializing into something similar to the above.

All the same I'll try using LocalDate and check what happens.